### PR TITLE
[front] - feat: toggle sidebar

### PR DESCRIPTION
**Pull request by JulesBelveze**
## Description

This PR aims at enabling toggle/untoggle the navigation bar on laptop: https://github.com/dust-tt/dust/issues/4988#issuecomment-2122616292

Screenshots of how it will look like:
Screenshot 2024-05-22 at 16 00 49
Screenshot 2024-05-22 at 16 00 27
Screenshot 2024-05-22 at 16 00 40

## Risk

Breaking UI

### CODE REVIEW

**Comment by JulesBelveze** [comment]
Screenshot 2024-05-22 at 16 00 49
Screenshot 2024-05-22 at 16 00 27
Screenshot 2024-05-22 at 16 00 40

**Review by JulesBelveze — COMMENTED** [review]
### Note
 Introduced a `NavigationBarContext` to mimic the `SidebarContext`. Curious to hear if this is no good practice?

### Question 
Why do we render a different nav bar for laptop and mobile? Why not a single instance deciding where to locate based on the device?

File: front/components/sparkle/AppLayout.tsx [hunk 1]
```diff
@@ -214,6 +282,7 @@ export default function AppLayout({
   children: React.ReactNode;
 }) {
   const { sidebarOpen, setSidebarOpen } = useContext(SidebarContext);
+  const { laptopNavOpen, setLaptopNavOpen } = useContext(NavigationBarContext);
```
Comment (spolu) [inline front/components/sparkle/AppLayout.tsx:285]: laptop?

File: front/components/assistant/conversation/input_bar/InputBar.tsx [hunk 2]
```diff
@@ -375,17 +375,19 @@ export function FixedAssistantInputBar({
...
+          <AssistantInputBar
+            owner={owner}
+            onSubmit={onSubmit}
+            conversationId={conversationId}
+            stickyMentions={stickyMentions}
+            additionalAgentConfiguration={additionalAgentConfiguration}
+            hideQuickActions={hideQuickActions}
+            disableAutoFocus={disableAutoFocus}
+          />
+        </div>
```
[the remark this answers is not in the document]
Comment (spolu) [reply front/components/assistant/conversation/input_bar/InputBar.tsx:390]: roger that will test :+1:

File: front/components/sparkle/AppLayout.tsx [hunk 3]
```diff
@@ -172,6 +216,30 @@ export const SidebarProvider = ({
   );
 };
 
+export const NavigationBarContext = React.createContext<{
```
Comment (spolu) [inline front/components/sparkle/AppLayout.tsx:219]: Why do we need a Provider? that seems very local to AppLayout no?

**Review by spolu — COMMENTED** [review]
Left a few comments. Will test locally.

Deferring review to @flvndvd as he has state/plans/ideas around the AppLayout

File: front/components/sparkle/AppLayout.tsx [hunk 4]
```diff
@@ -356,24 +425,29 @@ export default function AppLayout({
...
-                owner={owner}
-                subscription={subscription}
-                subNavigation={subNavigation}
-                topNavigationCurrent={topNavigationCurrent}
-              >
-                {loaded && navChildren && navChildren}
-              </NavigationBar>
-            )}
-          </div>
+          <Transition.Root show={laptopNavOpen} as={Fragment}>
```
Comment (spolu) [inline front/components/sparkle/AppLayout.tsx:428]: ooc what do we need a transition here? It's not really animated is it?

File: front/components/sparkle/AppLayout.tsx (hunk 1 above)
Comment (JulesBelveze) [reply front/components/sparkle/AppLayout.tsx:285]: That basically is the sidebar for non mobile devices, right? But agree naming can be better will change that 😄

File: front/components/sparkle/AppLayout.tsx (hunk 3 above)
Comment (JulesBelveze) [reply front/components/sparkle/AppLayout.tsx:219]: Yes that's why I was wondering if having a context here was relevant or not

Comment (spolu) [reply front/components/sparkle/AppLayout.tsx:219]: If you don't need it, don't use it :)

File: front/components/sparkle/AppLayout.tsx (hunk 4 above)
Comment (JulesBelveze) [reply front/components/sparkle/AppLayout.tsx:428]: Nop it's not. Mainly used it for ease of implementation.

Comment (JulesBelveze) [reply front/components/sparkle/AppLayout.tsx:428]: But will use conditional rendering 👍🏼

File: front/components/sparkle/AppLayout.tsx (hunk 3 above)
Comment (JulesBelveze) [reply front/components/sparkle/AppLayout.tsx]: Indeed, a simple state works 😸

File: front/components/sparkle/AppLayout.tsx [hunk 5]
```diff
@@ -427,6 +475,13 @@ export default function AppLayout({
             </div>
           </main>
         </div>
+
+        <div>
+          <ToggleSideBarButton
+            navigationBarOpen={navigationBarOpen}
+            setNavigationBarOpen={setNavigationBarOpen}
```
Comment (flvndvd) [inline front/components/sparkle/AppLayout.tsx:482]: To keep the state management within the main function component.
```suggestion
            toggleNavigationBarVisibility={(isVisible) => setNavigationBarOpen(isVisible)}
```

File: front/components/sparkle/AppLayout.tsx [hunk 6]
```diff
@@ -355,8 +400,11 @@ export default function AppLayout({
           </Transition.Root>
         )}
 
-        {!hideSidebar && (
-          <div className="hidden lg:fixed lg:inset-y-0 lg:z-0 lg:flex lg:w-80 lg:flex-col">
+        {!hideSidebar && navigationBarOpen && (
+          <div
+            id="nav-bar"
```
Comment (flvndvd) [inline front/components/sparkle/AppLayout.tsx:405]: ✂️

File: front/components/sparkle/AppLayout.tsx [hunk 7]
```diff
@@ -26,6 +40,36 @@ import { classNames } from "@app/lib/utils";
...
+    );
+  }, [navigationBarOpen, toggleNavigationBarVisibility]);
+
+  return (
+    <div
+      ref={buttonRef}
+      onClick={handleClick}
+      className={`hidden lg:fixed lg:top-1/2 lg:flex lg:h-10 lg:w-5 ${
+        navigationBarOpen ? "lg:left-80" : "lg:left-2"
+      }`}
```
Comment (flvndvd) [inline front/components/sparkle/AppLayout.tsx:66]: You can use the helper `classNames()`.

**Comment by flvndvd** [comment]
The width of the messages has changed: 
image

**Comment by JulesBelveze** [comment]
As per IRL discussion with @flvndvd, waiting for him to ship the new discussion design to rebase and use it.

**Comment by spolu** [comment]
What's the new discussion design?

**Comment by JulesBelveze** [comment]
With the rounded text boxes. Just tagged you in the corresponding Figma

**Comment by spolu** [comment]
Ah this PR! :+1: Roger :+1:

**Comment by JulesBelveze** [comment]
@flvndvd The only way I've found to fix the centering issue of the `FixedAssistantInputBar` is to depend on the state of the navigationBar and has something like:
```javascript
<div className={classNames("4xl:px-0 fixed bottom-0 left-0 right-0 z-20 flex-initial",
      isNavigationBarOpen? "lg:left-80": "")}>
      <div className="mx-auto max-h-screen max-w-4xl pb-0 sm:pb-8">
....
```
This seems to work but will require to create a `NavigationBar` context. What do you think?

**Comment by JulesBelveze** [comment]
As per IRL we will keep the current implementation, there's a bit of a glitch for viewports with a width in `[1024, 1200]px`

File: front/components/sparkle/AppLayout.tsx [hunk 8]
```diff
@@ -26,6 +40,36 @@ import { classNames } from "@app/lib/utils";
  */
 const SHOW_INCIDENT_BANNER = false;
 
+function ToggleSideBarButton({
+  navigationBarOpen,
+  toggleNavigationBarVisibility,
+}: {
+  navigationBarOpen: boolean;
+  toggleNavigationBarVisibility: (open: boolean) => void;
```
Comment (flvndvd) [inline front/components/sparkle/AppLayout.tsx:48]: `nit`
```suggestion
  toggleNavigationBarVisibility: (isOpened: boolean) => void;
```

File: front/components/sparkle/AppLayout.tsx [hunk 9]
```diff
@@ -214,6 +257,7 @@ export default function AppLayout({
   children: React.ReactNode;
 }) {
   const { sidebarOpen, setSidebarOpen } = useContext(SidebarContext);
+  const [navigationBarOpen, setNavigationBarOpen] = useState(true);
```
Comment (flvndvd) [inline front/components/sparkle/AppLayout.tsx:260]:
```suggestion
  const [isNavigationBarOpened, setIsNavigationBarOpened] = useState(true);
```

File: front/package.json [hunk 10]
```diff
@@ -18,6 +18,7 @@
     "@amplitude/analytics-browser": "^2.5.2",
     "@amplitude/analytics-node": "^1.3.5",
     "@auth0/nextjs-auth0": "^3.5.0",
+    "@dust-tt/sparkle": "0.2.152",
```
Comment (flvndvd) [inline front/package.json:21]: You have a duplicated version of sparkle here!

**Comment by albandum** [comment]
How do you open again the navbar from the first screenshot ?

**Comment by JulesBelveze** [comment]
@albandum There's a small button on the left hand side

**Comment by albandum** [comment]
> @albandum There's a small button on the left hand side

Ok - not visible from the screenshot (I guess I thought it was something on my screen haha) but if there's some sort of animation that works 👍

File: front/components/sparkle/AppLayout.tsx [hunk 11]
```diff
@@ -417,16 +462,18 @@ export default function AppLayout({
...
-            >
-              {loaded && children}
-            </div>
+            <div className="mx-auto h-full max-w-4xl">{loaded && children}</div>
           </main>
         </div>
+
+        <div>
+          <ToggleSideBarButton
+            navigationBarOpen={isNavigationBarOpened}
```
Comment (flvndvd) [inline front/components/sparkle/AppLayout.tsx:471]: `nit`

Could have been `isNavigationBarOpened` 😛

File: front/components/sparkle/AppLayout.tsx [hunk 12]
```diff
@@ -26,6 +40,36 @@ import { classNames } from "@app/lib/utils";
...
+    );
+  }, [navigationBarOpen, toggleNavigationBarVisibility]);
+
+  return (
+    <div
+      ref={buttonRef}
+      onClick={handleClick}
+      className={`hidden lg:fixed lg:top-1/2 lg:flex lg:w-5 ${
+        navigationBarOpen ? "lg:left-80" : "lg:left-0"
+      }`}
```
Comment (flvndvd) [inline front/components/sparkle/AppLayout.tsx:66]:
```suggestion
      className={classNames("hidden lg:fixed lg:top-1/2 lg:flex lg:w-5"
        navigationBarOpen ? "lg:left-80" : "lg:left-0"
      )}
```

**Review by flvndvd — COMMENTED** [review]
You need to remove the changes on pacakge-lock.json as well.

**Review by flvndvd — APPROVED** [review]

**Pull request merged**